### PR TITLE
Additions for BBH Domain creation

### PR DESCRIPTION
**Pull request by mar-celine**
## Proposed changes

Addition of tools necessary for BBH Domain creation.

### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- Follows [code review guidelines](https://sxs-collaboration.github.io/spectre/code_review_guide.html)
- Code has documentation and unit tests
- Private member variables have a trailing underscore
- Do not use [Hungarian notation](https://en.wikipedia.org/wiki/Hungarian_notation), e.g. `double* pd_blah` is bad
- Header order:
  1. hpp corresponding to cpp (only in cpp files)
  2. Blank line (only in cpp files)
  3. STL and externals (in alphabetical order)
  4. Blank line
  5. SpECTRE includes (in alphabetical order)
- File lists in CMake are alphabetical
- Correct `noexcept` specification for functions (if unsure, mark `noexcept`)
- Mark objects `const` whenever possible
- Almost always `auto`, except with expression templates, i.e. `DataVector`
- All commits for performance changes provide quantitative evidence and the tests used to obtain said evidence.
- Make sure error messages are helpful, e.g. "The number of grid points in the matrix 'F' is not the same as the number of grid points in the determinant."
- Prefix commits addressing PR requests with `fixup`. These commits are flagged
  by Travis so we do not accidentally merge them.
- Include what you use, prefer forward declarations in hpp files.
- Explicitly make numbers floating point, e.g. `2.` or `2.0` over `2`
- Use `Tensor`'s non-member get if the indices are constant expressions, e.g.
  `get<1>(tensor)`

### CODE REVIEW

File: src/Domain/DomainHelpers.cpp [hunk 1]
```diff
@@ -328,36 +332,83 @@ std::vector<std::unique_ptr<CoordinateMapBase<Frame::Logical, TargetFrame, 3>>>
 wedge_coordinate_maps(const double inner_radius, const double outer_radius,
                       const double inner_sphericity,
                       const double outer_sphericity,
-                      const bool use_equiangular_map) noexcept {
+                      const bool use_equiangular_map,
+                      const double x_coord_of_center) noexcept {
```
Comment (mar-celine) [inline src/Domain/DomainHelpers.cpp]: x_coord_of_shell_center

File: src/Domain/DomainHelpers.cpp [hunk 2]
```diff
@@ -328,36 +332,83 @@ std::vector<std::unique_ptr<CoordinateMapBase<Frame::Logical, TargetFrame, 3>>>
...
                       Direction<3>::upper_eta()}}),
-                 inner_sphericity, outer_sphericity, use_equiangular_map});
+                 inner_sphericity,
+                 outer_sphericity,
+                 use_equiangular_map};
+
+  if (x_coord_of_center != 0.0) {
+    using Identity2D = CoordinateMaps::Identity<2>;
+    using Affine = CoordinateMaps::Affine;
+    const auto shift_1d =
```
Comment (mar-celine) [inline src/Domain/DomainHelpers.cpp]: non-const, move into translation, move each of wedge_upper_z etc

File: src/Domain/DomainHelpers.hpp [hunk 3]
```diff
@@ -67,8 +67,12 @@ void set_periodic_boundaries(
 
 /// \ingroup ComputationalDomainGroup
 /// These are the six Wedge3Ds used in the DomainCreators for Sphere and Shell.
+/// The last argument `distance_to_plane` is used in the BBH DomainCreator to
+/// specify the distance from the center of the Shell to the plane between the
+/// two Shells.
```
Comment (mar-celine) [inline src/Domain/DomainHelpers.hpp]: x_coord_of_shell_center is a better name

File: src/Domain/DomainCreators/Shell.cpp [hunk 4]
```diff
@@ -45,8 +45,8 @@ Domain<3, TargetFrame> Shell<TargetFrame>::create_domain() const noexcept {
   std::vector<std::array<size_t, 8>> corners{
       {{7, 5, 8, 6, 15, 13, 16, 14}},   // Upper z
       {{4, 2, 3, 1, 12, 10, 11, 9}},    // Lower z
-      {{4, 8, 2, 6, 12, 16, 10, 14}},   // Upper y
```
Comment (mar-celine) [inline src/Domain/DomainCreators/Shell.cpp]: Edit commit message to reflect why these changes were made

File: src/Domain/DomainHelpers.cpp [hunk 5]
```diff
@@ -385,6 +385,84 @@ wedge_coordinate_maps(const double inner_radius, const double outer_radius,
...
+        Wedge3DMap{inner_radius,
+                   outer_radius,
+                   OrientationMap<3>(std::array<Direction<3>, 3>{
+                       {Direction<3>::upper_xi(), Direction<3>::upper_zeta(),
+                        Direction<3>::lower_eta()}}),
+                   inner_sphericity,
+                   outer_sphericity,
+                   use_equiangular_map,
+                   Halves::UpperOnly};
+    const auto wedge_lower_y_upper_half =
```
Comment (mar-celine) [inline src/Domain/DomainHelpers.cpp]: non const and move

File: src/Domain/DomainHelpers.cpp [hunk 6]
```diff
@@ -385,6 +385,84 @@ wedge_coordinate_maps(const double inner_radius, const double outer_radius,
                  outer_sphericity,
                  use_equiangular_map};
 
+  if (use_half_wedges) {
```
Comment (mar-celine) [inline src/Domain/DomainHelpers.cpp]: Edit documentation to explain which wedges are being cut

File: src/Domain/DomainHelpers.cpp [hunk 7]
```diff
@@ -327,6 +327,33 @@ void set_periodic_boundaries(
   }
 }
 
+std::array<OrientationMap<3>, 6> orientations_for_envelopes() {
```
Comment (mar-celine) [inline src/Domain/DomainHelpers.cpp]: noexcept

File: src/Domain/DomainHelpers.cpp [hunk 8]
```diff
@@ -335,132 +362,34 @@ wedge_coordinate_maps(const double inner_radius, const double outer_radius,
...
-                 outer_radius,
-                 OrientationMap<3>(std::array<Direction<3>, 3>{
-                     {Direction<3>::lower_zeta(), Direction<3>::lower_xi(),
-                      Direction<3>::upper_eta()}}),
-                 inner_sphericity,
-                 outer_sphericity,
-                 use_equiangular_map};
+  std::vector<Wedge3DMap> wedges{};
+  for (size_t i = 0; i < 6; i++) {
+    wedges.push_back(
```
Comment (mar-celine) [inline src/Domain/DomainHelpers.cpp]: use emplace_back and no need to write out Wedge3DMap

File: src/Domain/DomainHelpers.cpp [hunk 9]
```diff
@@ -335,132 +362,34 @@ wedge_coordinate_maps(const double inner_radius, const double outer_radius,
...
-                       {Direction<3>::upper_xi(), Direction<3>::lower_zeta(),
-                        Direction<3>::upper_eta()}}),
-                   inner_sphericity,
-                   outer_sphericity,
-                   use_equiangular_map,
-                   Halves::UpperOnly};
+    std::vector<Wedge3DMap> lower_wedges{};
+    std::vector<Wedge3DMap> upper_wedges{};
+    for (size_t i = 0; i < 4; i++) {
+      lower_wedges.push_back(
```
Comment (mar-celine) [inline src/Domain/DomainHelpers.cpp]: emplace back

File: src/Domain/DomainHelpers.cpp [hunk 10]
```diff
@@ -335,132 +362,34 @@ wedge_coordinate_maps(const double inner_radius, const double outer_radius,
...
+          Wedge3DMap{inner_radius, outer_radius, gsl::at(wedge_orientations, i),
+                     inner_sphericity, outer_sphericity, use_equiangular_map,
+                     Halves::LowerOnly});
+      upper_wedges.push_back(
+          Wedge3DMap{inner_radius, outer_radius, gsl::at(wedge_orientations, i),
+                     inner_sphericity, outer_sphericity, use_equiangular_map,
+                     Halves::LowerOnly});
```
Comment (mar-celine) [inline src/Domain/DomainHelpers.cpp]: upperonly

File: src/Domain/DomainHelpers.cpp [hunk 11]
```diff
@@ -471,21 +400,20 @@ wedge_coordinate_maps(const double inner_radius, const double outer_radius,
     const auto translation = CoordinateMaps::ProductOf2Maps<Affine, Identity2D>(
         shift_1d, Identity2D{});
     return make_vector(make_coordinate_map_base<Frame::Logical, TargetFrame>(
-                           wedge_upper_z, translation),
+                           wedges[0], translation),
```
Comment (mar-celine) [inline src/Domain/DomainHelpers.cpp]: std move is still possible here

File: src/Domain/DomainHelpers.cpp [hunk 12]
```diff
@@ -471,21 +400,20 @@ wedge_coordinate_maps(const double inner_radius, const double outer_radius,
...
-                           wedge_upper_x, translation),
+                           wedges[4], translation),
                        make_coordinate_map_base<Frame::Logical, TargetFrame>(
-                           wedge_lower_x, translation));
+                           wedges[5], translation));
   }
   return make_vector_coordinate_map_base<Frame::Logical, TargetFrame>(
-      wedge_upper_z, wedge_lower_z, wedge_upper_y, wedge_lower_y, wedge_upper_x,
-      wedge_lower_x);
+      wedges[0], wedges[1], wedges[2], wedges[3], wedges[4], wedges[5]);
```
Comment (mar-celine) [inline src/Domain/DomainHelpers.cpp]: move here as well

File: src/Domain/DomainHelpers.hpp [hunk 13]
```diff
@@ -69,10 +69,21 @@ void set_periodic_boundaries(
 /// These are the six Wedge3Ds used in the DomainCreators for Sphere and Shell.
 /// The last argument `distance_to_plane` is used in the BBH DomainCreator to
 /// specify the distance from the center of the Shell to the plane between the
-/// two Shells.
+/// two Shells. This function can also be used to produce the CoordinateMaps
+/// which envelope the Sphere and Shell in a cube made of six Wedges.
```
Comment (mar-celine) [inline src/Domain/DomainHelpers.hpp]: surround

File: src/Domain/DomainHelpers.cpp [hunk 14]
```diff
@@ -416,6 +417,58 @@ wedge_coordinate_maps(const double inner_radius, const double outer_radius,
...
+  const auto frustum_orientations = orientations_for_envelopes();
+  // In the BBH Domain, the cubes enveloping the Shells each have a side
+  // length of 2L.
+  const double L = frustum_parameter_l;
+  // In the BBH Domain, the frustums enveloping the aforementioned cubes
+  // make up a cube of their own, of length 2T.
+  const double T = frustum_parameter_t;
+
+  using FrustumMap = CoordinateMaps::Frustum;
+  // The BBH Domain uses 10 Frustums, 4 enveloping each Shell in the
```
Comment (mar-celine) [inline src/Domain/DomainHelpers.cpp]: around

File: src/Domain/DomainHelpers.cpp [hunk 15]
```diff
@@ -416,6 +417,58 @@ wedge_coordinate_maps(const double inner_radius, const double outer_radius,
...
+  //+z,-z,+y,-y directions, and 1 Frustum capping each end in +x, -x.
+  std::vector<FrustumMap> frustums_on_left{};
+  std::vector<FrustumMap> frustums_on_right{};
+  for (size_t i = 0; i < 4; i++) {
+    frustums_on_left.push_back(
```
Comment (gsb76) [inline src/Domain/DomainHelpers.cpp]: try emplace_back

File: src/Domain/DomainHelpers.cpp [hunk 16]
```diff
@@ -416,6 +417,58 @@ wedge_coordinate_maps(const double inner_radius, const double outer_radius,
...
+    frustums_on_right.push_back(
+        FrustumMap{{{{{0.0, -L}}, {{2 * L, L}}, {{0.0, -T}}, {{T, T}}}},
+                   L,
+                   T,
+                   gsl::at(frustum_orientations, i),
+                   use_equiangular_map});
+  }
+  frustums_on_left.push_back(
+      FrustumMap{{{{{-L, -L}}, {{L, L}}, {{-T, -T}}, {{T, T}}}},
+                 L,
```
Comment (mar-celine) [inline src/Domain/DomainHelpers.cpp]: 2L

File: src/Domain/DomainHelpers.cpp [hunk 17]
```diff
@@ -416,6 +417,58 @@ wedge_coordinate_maps(const double inner_radius, const double outer_radius,
...
+                 T,
+                 frustum_orientations[5],
+                 use_equiangular_map});
+  frustums_on_right.push_back(
+      FrustumMap{{{{{-L, -L}}, {{L, L}}, {{-T, -T}}, {{T, T}}}},
+                 L,
```
Comment (mar-celine) [inline src/Domain/DomainHelpers.cpp]: 2L

File: src/Domain/DomainCreators/RegisterDerivedWithCharm.cpp [hunk 18]
```diff
@@ -37,6 +39,8 @@ void register_with_charm<2>() {
                                                 CoordinateMaps::Equiangular>>));
   PUPable_reg(SINGLE_ARG(::CoordinateMap<Frame::Logical, Frame::Inertial,
                                          CoordinateMaps::Wedge2D>));
+  PUPable_reg(SINGLE_ARG(::CoordinateMap<Frame::Logical, Frame::Inertial,
```
Comment (kidder) [inline src/Domain/DomainCreators/RegisterDerivedWithCharm.cpp]: alphabetize

File: src/Domain/DomainHelpers.hpp [hunk 19]
```diff
@@ -66,9 +66,18 @@ void set_periodic_boundaries(
...
 
 /// \ingroup ComputationalDomainGroup
-/// These are the six Wedge3Ds used in the DomainCreators for Sphere and Shell.
+/// These are the Wedge3Ds used in the Sphere, Shell, and BBH DomainCreators.
+/// The last argument `x_coord_of_shell_center` is used in the BBH DomainCreator
+/// to specify the x-coordinate of the center of the Shell in the TargetFrame.
+/// This function can also be used to produce the CoordinateMaps
+/// which wrap the Sphere and Shell in a cube made of six Wedges.
+/// When `use_half_wedges` is `true`, the wedges in the +z,-z,+y,-y directions
+/// are cut in half along their xi-axes. These ten CoordinateMaps are used for
```
Comment (kidder) [inline src/Domain/DomainHelpers.hpp]: These -> the

File: src/Domain/DomainHelpers.cpp [hunk 20]
```diff
@@ -323,41 +327,105 @@ void set_periodic_boundaries(
...
+        std::move(wedges[4]), std::move(wedges[5]));             // NOLINT
+  }
+
+  if (x_coord_of_shell_center != 0.0) {
+    using Identity2D = CoordinateMaps::Identity<2>;
+    using Affine = CoordinateMaps::Affine;
+    auto shift_1d = Affine{-1.0, 1.0, -1.0 + x_coord_of_shell_center,
+                           1.0 + x_coord_of_shell_center};
+    const auto translation = CoordinateMaps::ProductOf2Maps<Affine, Identity2D>(
+        // clang-tidy: trivially copyable
```
Comment (kidder) [inline src/Domain/DomainHelpers.cpp]: move up one line

File: src/Domain/DomainHelpers.hpp [hunk 21]
```diff
@@ -81,3 +81,13 @@ wedge_coordinate_maps(double inner_radius, double outer_radius,
                       bool use_equiangular_map,
                       double x_coord_of_shell_center = 0.0,
                       bool use_half_wedges = false) noexcept;
+
+/// \ingroup ComputationalDomainGroup
+/// These are the ten Frustums used in the BBH DomainCreator. If the cubes
+/// enveloping the two Shells each have a side length of 2L, then the argument
+/// that should be passed to `frustum_parameter_l` should be L. The ten frustums
+/// make up a cube of their own, of side length 2 * `frustum_parameter_t`.
```
Comment (kidder) [inline src/Domain/DomainHelpers.hpp]: half_length_outer_cube and half_length_inner_cubes (and if easy, get rid of the halves)

File: src/Domain/DomainHelpers.cpp [hunk 22]
```diff
@@ -428,6 +429,65 @@ wedge_coordinate_maps(const double inner_radius, const double outer_radius,
...
+  // length of 2L.
+  const double L = frustum_parameter_l;
+  // In the BBH Domain, the frustums which surround the aforementioned cubes
+  // make up a cube of their own, of length 2T.
+  const double T = frustum_parameter_t;
+
+  using FrustumMap = CoordinateMaps::Frustum;
+  // The BBH Domain uses 10 Frustums, 4 around each Shell in the
+  //+z,-z,+y,-y directions, and 1 Frustum capping each end in +x, -x.
+  std::vector<FrustumMap> frustums_on_left{};
```
Comment (kidder) [inline src/Domain/DomainHelpers.cpp]: try just using one vector

**Review by kidder — CHANGES_REQUESTED** [review]

**Comment by kidder** [comment]
see if you can improve code coverage

File: src/Domain/DomainHelpers.cpp (hunk 20 above)
Comment (osheamonn) [reply src/Domain/DomainHelpers.cpp]: ping

File: src/Domain/DomainHelpers.cpp [hunk 23]
```diff
@@ -428,6 +429,62 @@ wedge_coordinate_maps(const double inner_radius, const double outer_radius,
...
+                                use_equiangular_map});
+  // frustum on the right
+  frustums.push_back(FrustumMap{{{{{-L, -L}}, {{L, L}}, {{-T, -T}}, {{T, T}}}},
+                                2.0 * L,
+                                T,
+                                frustum_orientations[4],
+                                use_equiangular_map});
+
+  return make_vector_coordinate_map_base<Frame::Logical, TargetFrame>(
+      // clang-tidy: trivially copyable
```
Comment (osheamonn) [inline src/Domain/DomainHelpers.cpp]: Move this up one line

File: src/Domain/DomainHelpers.cpp [hunk 24]
```diff
@@ -323,41 +328,161 @@ void set_periodic_boundaries(
...
+      lower_wedges.emplace_back(inner_radius, outer_radius,
+                                gsl::at(wedge_orientations, i),
+                                inner_sphericity, outer_sphericity,
+                                use_equiangular_map, Halves::LowerOnly);
+      upper_wedges.emplace_back(inner_radius, outer_radius,
+                                gsl::at(wedge_orientations, i),
+                                inner_sphericity, outer_sphericity,
+                                use_equiangular_map, Halves::UpperOnly);
+    }
+        // clang-tidy: trivially copyable
```
Comment (nilsdeppe) [inline src/Domain/DomainHelpers.cpp]: Please format this line

File: src/Domain/DomainHelpers.cpp [hunk 25]
```diff
@@ -323,41 +328,161 @@ void set_periodic_boundaries(
...
-                 OrientationMap<3>(std::array<Direction<3>, 3>{
-                     {Direction<3>::upper_zeta(), Direction<3>::upper_xi(),
-                      Direction<3>::upper_eta()}}),
-                 inner_sphericity, outer_sphericity, use_equiangular_map},
-      Wedge3DMap{inner_radius, outer_radius,
-                 OrientationMap<3>(std::array<Direction<3>, 3>{
-                     {Direction<3>::lower_zeta(), Direction<3>::lower_xi(),
-                      Direction<3>::upper_eta()}}),
-                 inner_sphericity, outer_sphericity, use_equiangular_map});
+      // clang-tidy: trivially copyable
```
Comment (nilsdeppe) [inline src/Domain/DomainHelpers.cpp]: move this up too.

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
clang-tidy wants you to sort the headers in Test_ConstGlobalCache.cpp

**Comment by kidder** [comment]
@gsb76 is this ready for final review?

**Comment by mar-celine** [comment]
@kidder yes this is ready :+1:

File: src/Domain/DomainHelpers.hpp [hunk 26]
```diff
@@ -66,9 +66,28 @@ void set_periodic_boundaries(
...
 /// \ingroup ComputationalDomainGroup
-/// These are the six Wedge3Ds used in the DomainCreators for Sphere and Shell.
+/// These are the Wedge3Ds used in the Sphere, Shell, and BBH DomainCreators.
+/// The last argument `x_coord_of_shell_center` is used in the BBH DomainCreator
+/// to specify the x-coordinate of the center of the Shell in the TargetFrame.
+/// This function can also be used to produce the CoordinateMaps
+/// which wrap the Sphere and Shell in a cube made of six Wedges.
+/// When `use_half_wedges` is `true`, the wedges in the +z,-z,+y,-y directions
+/// are cut in half along their xi-axes. The ten CoordinateMaps are used for
+/// the outermost Blocks of the BBH Domain.
```
Comment (fmahebert) [inline src/Domain/DomainHelpers.hpp]: Can I suggest reorganizing these sentences? I found the logical flow hard to follow.

I took a stab at reorganizing here, that (to me) is easier to follow. But it's possible I misunderstood what is happening and wrote incorrect stuff, so feel free to modify or even ignore:
```
/// These are the CoordinateMaps of the Wedge3Ds used in the Sphere, Shell, and
/// BBH DomainCreators.
/// This function can also be used to wrap the Sphere or Shell in a cube made of
/// six Wedges.
/// The argument `x_coord_of_shell_center` specifies a translation of the Shell
/// in the x direction in the TargetFrame. The BBH DomainCreator uses this to set
/// the position of each BH.
/// When the argument `use_half_wedges` is set to `true`, the wedges in the
/// +z,-z,+y,-y directions are cut in half along their xi-axes. The resulting ten
/// CoordinateMaps are used for the outermost Blocks of the BBH Domain.
```

Comment (mar-celine) [reply src/Domain/DomainHelpers.hpp]: I will use your reorganized sentences :+1:  thanks!

File: src/Domain/DomainHelpers.cpp [hunk 27]
```diff
@@ -323,41 +328,161 @@ void set_periodic_boundaries(
...
+  std::vector<Wedge3DMap> wedges{};
+  for (size_t i = 0; i < 6; i++) {
+    wedges.emplace_back(inner_radius, outer_radius,
+                        gsl::at(wedge_orientations, i), inner_sphericity,
+                        outer_sphericity, use_equiangular_map);
+  }
+
+  if (use_half_wedges) {
+    using Halves = Wedge3DMap::WedgeHalves;
+    std::vector<Wedge3DMap> lower_wedges{};
```
Comment (mar-celine) [inline src/Domain/DomainHelpers.cpp]: use a single vector if you can

File: src/Domain/DomainHelpers.cpp [hunk 28]
```diff
@@ -323,41 +328,161 @@ void set_periodic_boundaries(
...
-                 OrientationMap<3>(std::array<Direction<3>, 3>{
-                     {Direction<3>::upper_zeta(), Direction<3>::upper_xi(),
-                      Direction<3>::upper_eta()}}),
-                 inner_sphericity, outer_sphericity, use_equiangular_map},
-      Wedge3DMap{inner_radius, outer_radius,
-                 OrientationMap<3>(std::array<Direction<3>, 3>{
-                     {Direction<3>::lower_zeta(), Direction<3>::lower_xi(),
-                      Direction<3>::upper_eta()}}),
-                 inner_sphericity, outer_sphericity, use_equiangular_map});
+      std::move(wedges[0]), std::move(wedges[1]),  // NOLINT
```
Comment (mar-celine) [inline src/Domain/DomainHelpers.cpp]: try to fix this

File: src/Domain/DomainHelpers.cpp [hunk 29]
```diff
@@ -323,41 +328,161 @@ void set_periodic_boundaries(
...
+      std::move(wedges[5]));                       // NOLINT
+}
+
+template <typename TargetFrame>
+std::vector<std::unique_ptr<CoordinateMapBase<Frame::Logical, TargetFrame, 3>>>
+frustum_coordinate_maps(const double length_inner_cube,
+                        const double length_outer_cube,
+                        const bool use_equiangular_map) noexcept {
+  const auto frustum_orientations = orientations_for_wrappings();
+  const double L = 0.5 * length_inner_cube;
```
Comment (mar-celine) [inline src/Domain/DomainHelpers.cpp]: change to `inner` and `outer` as opposed to L and T

File: src/Domain/DomainHelpers.cpp [hunk 30]
```diff
@@ -399,10 +524,24 @@ template std::vector<
...
 wedge_coordinate_maps(const double inner_radius, const double outer_radius,
                       const double inner_sphericity,
                       const double outer_sphericity,
-                      const bool use_equiangular_map) noexcept;
+                      const bool use_equiangular_map,
+                      const double x_coord_of_shell_center,
+                      const bool use_wedge_halves) noexcept;
+template std::vector<
+    std::unique_ptr<CoordinateMapBase<Frame::Logical, Frame::Inertial, 3>>>
+frustum_coordinate_maps(const double frustum_parameter_l,
```
Comment (mar-celine) [inline src/Domain/DomainHelpers.cpp]: make names consistent with hpp

File: src/Domain/DomainHelpers.hpp [hunk 31]
```diff
@@ -66,9 +66,29 @@ void set_periodic_boundaries(
         neighbors_of_all_blocks);
 
 /// \ingroup ComputationalDomainGroup
-/// These are the six Wedge3Ds used in the DomainCreators for Sphere and Shell.
+/// These are the CoordinateMaps of the Wedge3Ds used in the Sphere, Shell, and
+/// BBH DomainCreators. This function can also be used to wrap the Sphere or
```
Comment (mar-celine) [inline src/Domain/DomainHelpers.hpp]: BBH -> Binary compact object

File: src/Domain/DomainHelpers.hpp [hunk 32]
```diff
@@ -66,9 +66,29 @@ void set_periodic_boundaries(
...
 wedge_coordinate_maps(double inner_radius, double outer_radius,
                       double inner_sphericity, double outer_sphericity,
-                      bool use_equiangular_map) noexcept;
+                      bool use_equiangular_map,
+                      double x_coord_of_shell_center = 0.0,
+                      bool use_half_wedges = false) noexcept;
+
+/// \ingroup ComputationalDomainGroup
+/// These are the ten Frustums used in the BBH DomainCreator. The cubes
+/// enveloping the two Shells each have a side length of `length_inner_cube`.
```
Comment (mar-celine) [inline src/Domain/DomainHelpers.hpp]: again

File: tests/Unit/Domain/Test_DomainHelpers.cpp [hunk 33]
```diff
@@ -127,3 +130,246 @@ SPECTRE_TEST_CASE("Unit.Domain.DomainHelpers.AllWedgeDirections",
   CHECK(*expected_coord_maps[4] == *maps[4]);
   CHECK(*expected_coord_maps[5] == *maps[5]);
 }
+
+SPECTRE_TEST_CASE("Unit.Domain.DomainHelpers.TranslatedSixWedgeDirections",
+                  "[Domain][Unit]") {
+  using Wedge3DMap = CoordinateMaps::Wedge3D;
+  const double inner_radius = 1.0;
```
Comment (mar-celine) [inline tests/Unit/Domain/Test_DomainHelpers.cpp]: dont use legendres constant

File: tests/Unit/Domain/Test_DomainHelpers.cpp [hunk 34]
```diff
@@ -127,3 +130,246 @@ SPECTRE_TEST_CASE("Unit.Domain.DomainHelpers.AllWedgeDirections",
...
+      make_coordinate_map_base<Frame::Logical, Frame::Inertial>(
+          Wedge3DMap{inner_radius, outer_radius,
+                     OrientationMap<3>{std::array<Direction<3>, 3>{
+                         {Direction<3>::lower_zeta(), Direction<3>::lower_xi(),
+                          Direction<3>::upper_eta()}}},
+                     inner_sphericity, outer_sphericity, use_equiangular_map},
+          translation));
+  const auto maps = wedge_coordinate_maps<Frame::Inertial>(
+      inner_radius, outer_radius, inner_sphericity, outer_sphericity,
+      use_equiangular_map, x_coord_of_shell_center);
```
Comment (mar-celine) [inline tests/Unit/Domain/Test_DomainHelpers.cpp]: factor out to test both equi angular and distant

File: tests/Unit/Domain/Test_DomainHelpers.cpp [hunk 35]
```diff
@@ -127,3 +130,246 @@ SPECTRE_TEST_CASE("Unit.Domain.DomainHelpers.AllWedgeDirections",
...
+  CHECK(*expected_coord_maps[7] == *maps[7]);
+  CHECK(*expected_coord_maps[8] == *maps[8]);
+  CHECK(*expected_coord_maps[9] == *maps[9]);
+}
+
+SPECTRE_TEST_CASE("Unit.Domain.DomainHelpers.AllFrustumDirections",
+                  "[Domain][Unit]") {
+  using FrustumMap = CoordinateMaps::Frustum;
+  // half of the length of the inner cube in the BBH domain:
+  const double L = 1.0;
```
Comment (mar-celine) [inline tests/Unit/Domain/Test_DomainHelpers.cpp]: change L and T here too

**Review by mar-celine — COMMENTED** [review]
squash

**Review by kidder — CHANGES_REQUESTED** [review]
squash

**Review by kidder — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]
